main.go

```diff
@@ -316,22 +308,30 @@ func main() {
 }
 
 func registerControllers(ctx context.Context, mgr manager.Manager) {
+	machineCache, err := coalescing.NewRequestCache(5 * time.Second)
+	if err != nil {
+		setupLog.Error(err, "failed to build machineCache ReconcileCache")
+	}
 	if err := controllers.NewAzureMachineReconciler(mgr.GetClient(), ctrl.Log.WithName("controllers").WithName("AzureMachine"),
 		mgr.GetEventRecorderFor("azuremachine-reconciler"),
 		reconcileTimeout,
 		watchFilterValue,
-	).SetupWithManager(ctx, mgr, controller.Options{MaxConcurrentReconciles: azureMachineConcurrency}); err != nil {
+	).SetupWithManager(ctx, mgr, controllers.Options{Options: controller.Options{MaxConcurrentReconciles: azureMachineConcurrency}, Cache: machineCache}); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "AzureMachine")
 		os.Exit(1)
 	}
 
+	clusterCache, err := coalescing.NewRequestCache(5 * time.Second)
```

Comment:
went with 5 seconds since I don't want the reconciles to be too slow but let me know if you think that's too aggressive